target/cheri-common/cheri-translate-utils.h

```diff
@@ -95,7 +95,15 @@ static inline void gen_check_pcc_bounds_next_inst(DisasContext *ctx,
     // case this assumption no longer holds in the future.
     tcg_debug_assert(ctx->base.pc_next >= ctx->base.pc_first);
     // XXX: __builtin_add_overflow() for end of address space?
-    if (unlikely(ctx->base.pc_next + num_bytes > ctx->base.pcc_top)) {
+    uint64_t ifetch_end_addr = 0;
+    if (unlikely(__builtin_add_overflow(ctx->base.pc_next, num_bytes,
+                                        &ifetch_end_addr))) {
+        warn_report("Found ifetch access that wraps around: " TARGET_FMT_lx
+                    " + %d.", ctx->base.pc_next, num_bytes);
+        cheri_tcg_prepare_for_unconditional_exception(&ctx->base);
+        gen_raise_pcc_violation(&ctx->base, ctx->base.pc_next, num_bytes);
```

Comment:
This isn't a trap. For MIPS, alignment ensures no instruction can straddle the boundary, so when wrapping each one is in bounds if base is zero. For RISC-V without C, the same applies, and with C, instructions are fetched as 16b sized and aligned parcels, so none of them can straddle, and our specification is that each is separately checked against PCC. Though, arguably, if you get into unaligned accesses, so long as every byte is in bounds it should be fine, regardless of whether you wrap.